receiver/mongodbatlasreceiver/events.go

```diff
@@ -0,0 +1,336 @@
+	}
+
+	er.record.NextStartTime = &et
+	return er.checkpoint(ctx)
+}
+
+func (er *eventsReceiver) poll(ctx context.Context, project *mongodbatlas.Project, p *ProjectConfig, startTime, now time.Time) {
+	for pageN := 1; pageN <= er.maxPages; pageN++ {
+		opts := &internal.GetEventsOptions{
+			PageNum:    pageN,
+			EventTypes: er.cfg.Events.Types,
+			MaxDate:    now,
+			MinDate:    startTime,
+		}
+
+		projectEvents, hasNext, err := er.client.GetEvents(ctx, project.ID, opts)
+		if err != nil {
+			er.logger.Error("unable to get events for project", zap.Error(err), zap.String("project", p.Name))
+			break
+		}
+
+		now := pcommon.NewTimestampFromTime(now)
+		logs := er.transformEvents(now, projectEvents, project)
+
+		if logs.LogRecordCount() > 0 {
+			if err = er.consumer.ConsumeLogs(ctx, logs); err != nil {
+				er.logger.Error("error consuming events", zap.Error(err))
+				break
+			}
+		}
+
+		if !hasNext {
+			break
+		}
+	}
+}
+
+func (er *eventsReceiver) transformEvents(now pcommon.Timestamp, events []*mongodbatlas.Event, p *mongodbatlas.Project) plog.Logs {
+	logs := plog.NewLogs()
+	resourceLogs := logs.ResourceLogs().AppendEmpty()
+	ra := resourceLogs.Resource().Attributes()
+	ra.PutStr("mongodbatlas.project.name", p.Name)
+	ra.PutStr("mongodbatlas.org.id", p.OrgID)
+
+	for _, event := range events {
+
+		logRecord := resourceLogs.ScopeLogs().AppendEmpty().LogRecords().AppendEmpty()
+		bodyBytes, err := json.Marshal(event)
+		if err != nil {
+			er.logger.Error("unable to unmarshal event into body string", zap.Error(err))
+			continue
+		}
+		logRecord.Body().SetStr(string(bodyBytes))
+
+		// ISO-8601 formatted
+		ts, err := time.Parse(time.RFC3339, event.Created)
+		if err != nil {
+			er.logger.Warn("unable to interpret when an event was created, expecting a RFC3339 timestamp", zap.String("timestamp", event.Created), zap.String("event", event.ID))
+			logRecord.SetTimestamp(now)
+		} else {
+			logRecord.SetTimestamp(pcommon.NewTimestampFromTime(ts))
+		}
+		logRecord.SetObservedTimestamp(now)
+
+		attrs := logRecord.Attributes()
+		// always present attributes
+		attrs.PutStr("event.domain", "mongodbatlas")
+		attrs.PutStr("type", event.EventTypeName)
+		attrs.PutStr("id", event.ID)
+		attrs.PutStr("group.id", event.GroupID)
+
+		parseOptionalAttributes(&attrs, event)
+	}
+	return logs
+}
+
+func (er *eventsReceiver) checkpoint(ctx context.Context) error {
+	marshalBytes, err := json.Marshal(er.record)
+	if err != nil {
+		return fmt.Errorf("unable to write checkpoint: %w", err)
+	}
+	return er.storageClient.Set(ctx, eventStorageKey, marshalBytes)
+}
+
+func (er *eventsReceiver) loadCheckpoint(ctx context.Context) {
+	cBytes, err := er.storageClient.Get(ctx, eventStorageKey)
+
+	if cBytes == nil {
+		er.record = &eventRecord{}
+		return
+	}
+
+	if err != nil {
+		er.logger.Info("unable to load checkpoint from storage client, continuing without a previous checkpoint", zap.Error(err))
+		er.record = &eventRecord{}
+		return
+	}
```

Comment:
Typically check the error before any other return value, so move this block above the previous one.